### PR TITLE
Change default KL estimator from k3 to k2 for loss-based KL

### PR DESCRIPTION
**Pull request by taivu1998**
## Summary

Addresses #805 by updating SkyRL's default KL estimator from `k3` to `k2` while keeping the existing default KL placement unchanged.

This PR intentionally keeps:
- `trainer.algorithm.use_kl_loss = true`
- `trainer.algorithm.use_kl_in_reward = false`

That makes the change narrow and low-risk: it modernizes the default estimator without also flipping the repo-wide default from loss-based KL to reward-based KL.

## What Changed

- Changed the default KL estimator to `k2` in the typed config dataclass.
- Updated `ppo_base_config.yaml` to match the Python config default.
- Updated the configuration docs to say `k2` is the default for loss-based KL and clarified that reward-based KL is still supported.
- Added a focused regression test for the default KL configuration.
- Updated the fallback default on `compute_approx_kl(...)` to `k2` for consistency with the new repo default.

## Why This Shape

Issue #805 discusses two possible directions:
- `k1` in reward
- `k2` in loss

This PR takes the `k2`-in-loss path because it aligns with the issue while preserving SkyRL's current default training behavior. It avoids turning this fix into a larger redesign of KL placement across the codebase.

## Files Changed

- `skyrl/train/config/config.py`
- `skyrl/train/config/ppo_base_config.yaml`
- `skyrl/backends/skyrl_train/utils/ppo_utils.py`
- `docs/content/docs/configuration/config.mdx`
- `tests/train/test_config.py`

## Validation

Directly relevant checks passed locally:

```bash
RAY_ENABLE_UV_RUN_RUNTIME_ENV=0 /Users/vuductai/Documents/Projects/SkyRL/.venv/bin/python -m pytest --noconftest tests/train/test_config.py::test_default_kl_regularization_config tests/backends/skyrl_train/utils/test_ppo_utils.py::test_compute_approx_kl -q
```

A broader non-Ray-gated pass also succeeded:

```bash
RAY_ENABLE_UV_RUN_RUNTIME_ENV=0 /Users/vuductai/Documents/Projects/SkyRL/.venv/bin/python -m pytest --noconftest tests/train/test_config.py tests/backends/skyrl_train/utils/test_ppo_utils.py -q -k 'not test_registry_cross_ray_process and not test_registry_named_actor_creation and not test_registry_reset_after_ray_shutdown'
```

Result:
- `35 passed, 3 deselected`

The three deselected tests are Ray registry tests that still require `ray.init()` process inspection, which is sandbox-restricted in this local macOS environment. CI should cover those normally.

## Notes

- This PR does not change `kl_loss_coef`.
- This PR does not change the default KL placement.
- Explicit user overrides of KL settings continue to behave the same way.

---

<a href="https://app.devin.ai/review/novasky-ai/skyrl/pull/1445" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

**Comment by tamoghnokandar** [comment]
@erictang000 Cursor's Composer 2 uses k1 as the estimator for KL divergence in the loss function. Reason being k3 has high variance and k2 is biased. Maybe we should use k1 then?


image
